pandas/core/frame.py

```diff
@@ -641,6 +641,8 @@ def __init__(
 
         if isinstance(data, DataFrame):
             data = data._mgr
+            if not copy:
+                data = data.copy(deep=False)
```

Comment:
comment about why?  otherwise in a few months we're gonna get the bright idea to slightly improve perf by removing this

Reply:
> we're gonna get the bright idea to slightly improve perf by removing this

The added tests should fail then and prevent us from doing so! ;) 
But yes, certainly, will add a comment.